src/rules/utils.ts

```diff
@@ -758,3 +758,32 @@ export const scopeHasLocalReference = (
     !references.unresolved.has(referenceName)
   );
 };
+
+/**
+ * Checks if node names returned by getNodeName matches any of the given star patterns
+ * Pattern examples:
+ *   request.*.expect
+ *   request.**.expect
+ *   request.**.expect*
+ */
+
+export function nodeNameMatch(
+  nodeName: string,
+  ...patterns: readonly string[]
```

Comment:
🐘 There's no real gain here to using rest, since I don't expect we'll be calling this in non-array conditions, and rest is requiring you to spread an array (effectively you're iterating over an array to extract elements out to be collected back into an array).